src/xb-value-bindings.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2020 Endless OS Foundation LLC
+ *
+ * Author: Philip Withnall <withnall@endlessm.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#pragma once
+
+#include <glib.h>
+
+#include "xb-opcode.h"
+
+G_BEGIN_DECLS
+
+/* TODO Docs */
+typedef struct {
+	gint dummy0;
+	gpointer dummy1[2];
+	gint dummy2;
+	gpointer dummy3[2];
+	gpointer dummy4[3];
+} XbValueBindings;
+
+/* TODO docs */
+#define XB_VALUE_BINDINGS_INIT() { 0, { NULL, NULL }, 0, { NULL, NULL }, { NULL, } }
+
+void		 xb_value_bindings_init		(XbValueBindings	 *self);
```

Comment:
I'm not sure I like the _name_. It's a mapping of bound indexes to values, right? More generally, it's thread global context for the query, so maybe something with `Context` in the name?

Reply:
I haven’t been able to come up with a better name. It is a mapping of bound indices to values, but I don’t think it’s as general as a context for the query. It doesn’t contain any other contextual data. Subsets of it are copied to new `XbValueBindings` instances for different predicates in the query (see the changes to `xb_silo_query_node_matches()` in “xb-query: Use XbValueBindings in query functions”), which wouldn’t work as well semantically if it was a ‘query context’.

Reply:
> I haven’t been able to come up with a better name

Me either. Lets stick to this for now.